src/main/java/amazon/DealsOfTodayPage.java

```diff
@@ -61,12 +60,19 @@ public boolean anyOfTheItemsIsSelected() {
 
     }
 
-    public void clickItemWhichHaveTextFashionOfWomenFromRitual() {
-        itemWhichHaveATextFashionOfWomenFromDailyRitual.click();
+    public void clickTheElementThatHasTheFollowingText(String name) {
+
+        try {
+            driver.findElement(By.linkText(name)).click();
+
+        } catch (Exception e) {
+
+            System.out.println("The requested product is not available on the page");
+        }
     }
```

Comment:
1. Change the method name to a understandable one, which will tell what does this method do actually.
2. Why you didn't create the element via FindBy annotation?
3. What does give you `System.out.println` in `catch` block? Your method gets terminated in case of exception appearance or it continues the execution? If it gets terminated, then all is OK, but will it be terminated??